components/homme/src/theta-l_kokkos/prim_driver_mod.F90

```diff
@@ -475,6 +475,7 @@ subroutine prim_run_subcycle(elem, hybrid, nets, nete, dt, single_column, tl, hv
     ! Print some diagnostic information
 
     if (compute_diagnostics) then
+       call sync_diagnostics_to_host_c()
        call prim_printstate(elem, tl, hybrid,hvcoord,nets,nete)
```

Comment:
This is now the only place a deep_copy() is done, right before diagnostics are output to screen.